### DISCUSSION TITLE
Code Highlight Widget - PowerShell support

### DISCUSSION THREAD

**Discussion by itsChris** [elementor/elementor#22886] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

When using the Pro Widget 'Code Highlight', PowerShell as Language is not supported

### Describe the solution you'd like

Code highlighting for PowerShell

### Describe alternatives you've considered

Coder
–
Code Syntax Highlighter for Elementor

it's 10 Euros/Dollars again. it seems it is no longer maintained/supported.

**Comment by NoralK** [comment]
I have no idea why no one at Elementor has not responded to this.  It is very simple, Edit the file Elementor-pro->modules->code-highlight->wigets->code-highlight.php.  You will set about line 90 a list of language options.  Insert this line `'powershell' => 'PowerShell',`.  The list should look something like this now:

``` php
		$language_option = [
			'markup' => 'Markup',
			'html' => 'HTML',
			'css' => 'CSS',
			'sass' => 'Sass (Sass)',
			'scss' => 'Sass (Scss)',
			'less' => 'Less',
			'javascript' => 'JavaScript',
			'typescript' => 'TypeScript',
			'jsx' => 'React JSX',
			'tsx' => 'React TSX',
			'php' => 'PHP',
			'ruby' => 'Ruby',
			'json' => 'JSON + Web App Manifest',
			'http' => 'HTTP',
			'xml' => 'XML',
			'svg' => 'SVG',
			'rust' => 'Rust',
			'csharp' => 'C#',
			'dart' => 'Dart',
			'git' => 'Git',
			'java' => 'Java',
			'sql' => 'SQL',
			'go' => 'Go',
			'kotlin' => 'Kotlin + Kotlin Script',
			'julia' => 'Julia',
			'python' => 'Python',
			'swift' => 'Swift',
			'bash' => 'Bash + Shell',
			'scala' => 'Scala',
			'haskell' => 'Haskell',
			'perl' => 'Perl',
			'objectivec' => 'Objective-C',
			'visual-basic,' => 'Visual Basic + VBA',
			'r' => 'R',
			'c' => 'C',
			'cpp' => 'C++',
			'aspnet' => 'ASP.NET (C#)',
			'powershell' => 'PowerShell',
		];
```

Save the file and you will now have an option for PowerShell.  This will work for any PrismJS supported language - https://prismjs.com/#supported-languages.
